compiler/codegen/stmt.cpp

```diff
@@ -142,7 +176,34 @@ GenRet BlockStmt::codegen() {
 
     info->lvt->addLayer();
 
-    body.codegen("");
+    // Add a new variables set to the stack for lifetimes
+    info->currentStackVariables.emplace_back();
+
+    for_alist(node, this->body) {
+      node->codegen();
+      if (DefExpr* def = toDefExpr(node)) {
+        Symbol* var = def->sym;
+        if (var->type && var->type != dtVoid && var->type != dtNothing) {
+          GenRet ret = var->codegen();
+          if (!isGlobal(var) && ret.isLVPtr == GEN_PTR) {
```

Comment:
Let's not run `var->codegen()` if `isGlobal(var)`. One way to handle it is to put that check in the previous conditional.